edx_exams/apps/api/v1/views.py

```diff
@@ -385,6 +385,10 @@ def get_response(cls, exam, user):
         elif exam.due_date is not None and timezone.now() >= exam.due_date:
             grant_access, response_status = True, status.HTTP_200_OK
 
+        # If user is course staff, then grant them access
```

Comment:
This is what we want, but I think we should have this before line 369. On the frontend, [access to the contents of the exam will always be granted to staff members](https://github.com/openedx/frontend-lib-special-exams/blob/4878aef9188a4b5df42ff0a2229b5d34a3624d09/src/exam/ExamWrapper.jsx#L45-L50). Because of this, they never see any exam functionality and will not be able to start an exam attempt. Therefore, we can short circuit early and give them access immediately if they are a staff member - no need to look at exam attempts. This makes the code easier to reason about. Does that match your understanding?